pkg/source/source.go

```diff
@@ -359,7 +415,14 @@ func (is *Informer) Start(ctx context.Context, handler handler.EventHandler, que
 		return fmt.Errorf("must specify Informer.Informer")
 	}
 
-	_, err := is.Informer.AddEventHandler(internal.EventHandler{Queue: queue, EventHandler: handler, Predicates: prct})
+	is.mu.Lock()
+	defer is.mu.Unlock()
+	if is.canceled {
+		return nil
```

Comment:
Shouldn't we error rather than silently doing nothing?

Reply:
I'm not sure, maybe we have to prevent the source been stopped multiple times by controller or user?
For example, someone may create a controller with three watches, then stop one watch manually, and finally stop the whole controller. At the time, controller will stop the stopped watch source and it will get the error.